src/components/Editor.jsx

```diff
@@ -0,0 +1,93 @@
+"use strict";
+
+var React = require('react');
+var ACE = require('brace');
+require('brace/mode/html');
+require('brace/mode/css');
+require('brace/mode/javascript');
+require('brace/theme/monokai');
+
+var ProjectActions = require('../actions/ProjectActions');
+var ProjectStore = require('../stores/ProjectStore');
+var ErrorStore = require('../stores/ErrorStore');
+
+var Editor = React.createClass({
+  componentDidMount: function(containerElement) {
+    this._setupEditor(containerElement);
+    ProjectStore.addChangeListener(this._onChange);
+    ErrorStore.addChangeListener(this._onChange);
+  },
+
+  componentDidUnmount: function() {
+    this.editor.destroy();
+    ProjectStore.removeChangeListener(this._onChange);
+    ErrorStore.removeChangeListener(this._onChange);
+  },
+
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.projectKey !== this.props.projectKey) {
+      this._startNewSession(this._getSource(nextProps.projectKey));
+    }
+  },
+
+  shouldComponentUpdate: function(nextProps) {
+    return false;
+  },
+
+  render: function() {
+    return (
+      <div className="editor">
+        {this._getSource()}
+      </div>
+    )
+  },
+
+  _jumpToLine: function(line, column) {
+    this.editor.moveCursorTo(line, column);
+    this.editor.focus();
+  },
+
+  _setupEditor: function(containerElement) {
+    this.editor = ACE.edit(this.getDOMNode());
+    this._configureSession(this.editor.getSession());
+  },
+
+  _onChange: function() {
```

Comment:
It’s weird that we are directly messing with the editor’s state here rather than using component state to mediate it (or maybe using component state would just be unnecessary indirection)